Objects/bytearrayobject.c

```diff
@@ -2391,11 +2391,17 @@ PyDoc_STRVAR(length_hint_doc,
 static PyObject *
 bytearrayiter_reduce(bytesiterobject *it, PyObject *Py_UNUSED(ignored))
 {
+    PyObject *iter = _PyEval_GetBuiltin(&_Py_ID(iter));
+
+    /* _PyEval_GetBuiltin can invoke arbitrary code.
+     * calls must be *before* access of `it` pointers,
+     * since C parameter eval order is undefined.
```

Comment:
I think parameter eval order isn't the issue and shouldn't be mentioned. The problem is that the `_PyEval_GetBuiltin` call invalidates the previous `if` check.

For example, this code would also be buggy:
```
    if (it->it_seq != NULL) {
        PyObject *iter = _PyEval_GetBuiltin(&_Py_ID(iter));
        return Py_BuildValue("N(O)n", iter, it->it_seq, it->it_index);
    }
```

The parameter evaluation order makes things a bit more unpredictable, but the bug is that the `if` check is done too early, not that we're relying on evaluation order.